code/game/machinery/kitchen/gibber.dm

```diff
@@ -295,18 +248,43 @@ obj/machinery/gibber/New()
 		src.operating = 0
 		update_icon()
 
-/obj/machinery/gibber/proc/startautogibbing(mob/living/victim as mob)
-	if(src.operating)
+
+
+//auto-gibs anything that bumps into it
+/obj/machinery/gibber/autogibber
+	name = "autogibber"
+	desc = "Keep far, far away."
+	icon_state = "autogibber"
+
+/obj/machinery/gibber/autogibber/New()
+	..()
+	overlays = null
+
+/obj/machinery/gibber/autogibber/attack_hand(mob/user as mob)
+	Bumped(user)
+
+/obj/machinery/gibber/autogibber/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	Bumped(user)
+
+/obj/machinery/gibber/autogibber/Bumped(var/atom/A)
+	if(stat & (BROKEN | NOPOWER))
 		return
+	use_power(100)
+	if(isliving(A))
+		var/mob/living/M = A
+		M.visible_message("<span class='warning'>[M] is forcefully sucked into \the [src]!</span>", \
+			drugged_message = "<span class='warning'>[M] suddenly vanishes! How did \he do that?</span>")
+		M.forceMove(src)
+		startautogibbing(M)
+
+/obj/machinery/gibber/autogibber/proc/startautogibbing(mob/living/victim as mob)
```

Comment:
The spawn is controlled by a configurable time, the user is only used for lastassailant - which should be handled outside of such a proc anyways - and it's human-only because it makes human meat, which startautogibbing also does.

Given the sheer amount of overlap, you should really be looking to merge these together.